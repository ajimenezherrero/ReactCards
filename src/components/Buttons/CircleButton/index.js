import styled from 'styled-components';
import PropTypes from 'prop-types';

const getComputedStyles = ({ backgroundColor, color, fontSize, size }) => (`
  height: ${size};
  width: ${size};
  background-color: ${backgroundColor};
  color: ${color};
  font-size: ${fontSize};
`)

const CircleButton = styled.button`
  display:block;
  border-radius: 50%;
  outline: none;
  cursor: pointer;
  border: none;

  ${props => getComputedStyles(props)}
`;

CircleButton.defaultProps = {
  size: '75px',
  color: 'white',
  backgroundColor: 'red',
  fontSize: '24px'
};

CircleButton.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  fontSize: PropTypes.string
};

export default CircleButton;
