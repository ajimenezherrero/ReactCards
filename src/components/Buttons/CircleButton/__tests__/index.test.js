import React from 'react';
import { shallow } from 'enzyme';


import CircleButton from '../index';

const mockComponent = props => <CircleButton { ...props }/>;

describe('<CircleButton />', () => {
  it('should render component', () => {
    const tree = shallow(mockComponent());

    expect(tree).toMatchSnapshot();
  });
});
