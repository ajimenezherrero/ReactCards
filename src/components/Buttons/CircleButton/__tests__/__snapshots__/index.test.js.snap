// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleButton /> should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.button
    backgroundColor="red"
    color="white"
    fontSize="24px"
    size="75px"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa fzLuEn",
      "color": "white",
      "fontSize": "24px",
      "size": "75px",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa fzLuEn",
        "color": "white",
        "fontSize": "24px",
        "size": "75px",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
